docs/src/getting_started/core.md

```diff
@@ -0,0 +1,209 @@
+   crux_core = "0.2.0"
+   serde = { version = "1.0.147", features = ["derive"] }
+   bincode = "1.3.3"
+
+   [build-dependencies]
+   uniffi_build = { version = "0.21.0", features = ["builtin-bindgen"] }
+   ```
+
+1. Create `./shared/src/shared.udl`
+
+   ```txt
+   namespace shared {
+     sequence<u8> message([ByRef] sequence<u8> msg);
+     sequence<u8> response([ByRef] sequence<u8> res);
+     sequence<u8> view();
+   };
+   ```
+
+1. Create `./shared/uniffi.toml`
+
+   ```toml
+   [bindings.kotlin]
+   package_name = "com.redbadger.rmm.shared"
+   cdylib_name = "shared"
+
+   [bindings.swift]
+   cdylib_name = "shared_ffi"
+   omit_argument_labels = true
+   ```
+
+1. Include the scaffolding in `./shared/src/lib.rs`
+
+   ```rust
+    pub mod app;
+
+    use lazy_static::lazy_static;
+    use wasm_bindgen::prelude::wasm_bindgen;
+
+    use crux_core::Core;
+    pub use crux_core::Request;
+    pub use crux_http as http;
+
+    pub use app::*;
+
+    uniffi_macros::include_scaffolding!("shared");
+
+    lazy_static! {
+        static ref CORE: Core<Effect, App> = Core::new::<Capabilities>();
+    }
+
+    # [wasm_bindgen]
+    pub fn message(data: &[u8]) -> Vec<u8> {
+        CORE.message(data)
+    }
+
+    # [wasm_bindgen]
+    pub fn response(uuid: &[u8], data: &[u8]) -> Vec<u8> {
+        CORE.response(uuid, data)
+    }
+
+    # [wasm_bindgen]
+    pub fn view() -> Vec<u8> {
+        CORE.view()
+    }
+   ```
+
+2. Create a basic app implementation in `./shared/src/app.rs`
+
+   ```rust
+   use crux_core::render::Render;
+   use crux_macros::Effect;
+   use serde::{Deserialize, Serialize};
+   
+   #[derive(Default)]
+   pub struct App;
+   
+   impl crux_core::App for App {
+       type Model = Model;
+       type Event = Event;
+       type ViewModel = ViewModel;
+       type Capabilities = Capabilities;
+   
+       fn update(&self, msg: Self::Event, model: &mut Self::Model, caps: &Self::Capabilities) {
+           match msg {
+               Event::Increment => {
+                   model.count += 1;
+                   caps.render.render();
+               },
+               Event::Decrement => {
+                   model.count -= 1;
+                   caps.render.render();
+               }
+           }
+       }
+   
+       fn view(&self, model: &Self::Model) -> Self::ViewModel {
+           model.into()
+       }
+   }
+   
+   #[derive(Default)]
+   pub struct Model {
+       count: isize,
+   }
+   
+   #[derive(Serialize, Deserialize)]
+   pub struct ViewModel {
+       pub text: String,
+   }
+   
+   impl From<&Model> for ViewModel {
+       fn from(model: &Model) -> Self {
+           Self {
+               text: model.count.value.to_string() + &suffix,
+           }
+       }
+   }
+   
+   #[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+   pub enum Event {
+       Increment,
+       Decrement,
+   }
+   
+   #[derive(Effect)]
+   pub struct Capabilities {
+       pub render: Render<Event>,
+   }
+   ```
```

Comment:
Not one for now, but might be worth just including the example files here?  https://rust-lang.github.io/mdBook/format/mdbook.html#including-files

Reply:
Potentially, although they all have extra stuff to explain, that's why I didn't. Maybe we need a dumb hello world example to point to for the basic scaffolding.

Reply:
I'll write your hello world as an example in a separate PR

Reply:
The example is in this [PR](https://github.com/redbadger/crux/pull/37/files). The order in which the example is built up is tricky because the `Effect` derive macro will fail until the rest of the example is in place.

Reply:
I updated the code in this tutorial to match the (as yet unmerged) example, so hopefully should be OK

Reply:
Once the other PR is merged, we could include the whole example in this page as per @obmarg's excellent suggestion above.